src/data/markdown/docs/02 javascript api/04 k6-data/1-SharedArray.md

```diff
@@ -28,7 +28,8 @@ Which means that for the most part if you currently have an array data structure
 import { SharedArray } from 'k6/data';
 
 const data = new SharedArray('some name', function () {
-  // here you can open files, and then do additional processing on them or just generate the data dynamically
+  // here you can generate the data dynamically or open files and do additional processing
+  // it's important to open large files inside this function to avoid memory saturation
```

Comment:
```suggestion
  // All heavy work (opening and processing big files for example) should be done inside here.
  // This way it will happen only once and the result will be shared between all VUs, saving time and memory.
```

Reply:
I feel like explaining *why* that is the case is equally if not important as I would expect that will be what the user will remember/understand  better.